lib/master.js

```diff
@@ -0,0 +1,189 @@
+/**
+ * @file In order to load huge volumes of polygon data into memory without
+ *    breaking Node (due to its low heap-size limits), the package forks a
+ *    worker process per polygon layer/shapefile. This module contains
+ *    functions for initializing them and searching them.
+ */
+
+'use strict';
+
+var childProcess = require( 'child_process' );
+var peliasConfig = require( 'pelias-config' );
+var path = require( 'path' );
+var through = require( 'through2' );
+var logger = require( 'pelias-logger' ).get( 'admin-lookup:master' );
+
+/**
+ * For every desired administrative layer, specifies its name, the partial path
+ * of the corresponding Quattroshapes file, and the properties to extract from
+ * it.
+ */
+var quattroAdminLevels = [
+  {
+    name: 'admin1',
+    path: 'adm1',
+    props: [ 'qs_adm0', 'qs_adm0_a3', 'qs_a1' ]
+  },
+  {
+    name: 'admin2',
+    path: 'adm2',
+    props: [ 'qs_a2' ]
+  },
+  {
+    name: 'local_admin',
+    path: 'localadmin',
+    props: [ 'qs_la' ]
+  },
+  {
+    name: 'locality',
+    path: 'localities',
+    props: [ 'qs_loc' ]
+  },
+  {
+    name: 'neighborhood',
+    path: 'neighborhoods',
+    props: [
+      'name', 'name_adm0', 'name_adm1', 'name_adm2', 'name_lau', 'name_local'
+    ]
+  }
+];
+
+/**
+ * Fork a `./worker.js` per layer in `quattroAdminLevels`, load the
+ * corresponding polygons into it, and pass all workers to the `cb` callback
+ * once finished.
+ */
+function initWorkers( cb ){
+  logger.info( 'Initializing workers.' );
+
+  var numLoadedLevels = 0;
+
+  var config = peliasConfig.generate();
+  var quattroPath = config.imports.quattroshapes.datapath;
+  var workers = [];
+  quattroAdminLevels.forEach( function ( lvlConfig ){
+    var worker = childProcess.fork( path.join( __dirname, 'worker' ) );
+    worker.on('message', function (){
+      if( ++numLoadedLevels === quattroAdminLevels.length ){
+        logger.info( 'Finished initializing workers.' );
+        cb( workers );
+      }
+    });
+    workers.push( worker );
+    lvlConfig.path = path.join( quattroPath, 'qs_' + lvlConfig.path );
```

Comment:
Of course, it might be interesting to allow the package consumer to specify the shapefile path. This would allow loading a single country (as we carve up Quattroshapes into country-specific extracts), as opposed to the planet.